flang/lib/Lower/RTBuilder.h

```diff
@@ -207,6 +207,20 @@ constexpr TypeBuilderFunc getModel<bool &>() {
   };
 }
 template <>
+constexpr TypeBuilderFunc getModel<std::complex<float> &>() {
+  return [](mlir::MLIRContext *context) -> mlir::Type {
+    return fir::ReferenceType::get(
+        fir::ComplexType::get(context, sizeof(float)));
```

Comment:
This should probably use the KindMap to figure out what KIND value corresponds to a `float` LLVM semantics rather than be hard-coded to whatever the build machine happens to be.

Reply:
This is now improved.